### PR TITLE
feat: customize `logits_soft_cap` value

### PR DESCRIPTION
**Pull request by yzh119**
This PR supports customized logits soft cap values. Different models might use different logits soft cap values (e.g. Grok-1 uses 30 and Gemma-2 uses 50).

### CODE REVIEW

**Comment by zhyncs** [comment]
> Gemma-2 uses 50

Great work!
